lms/djangoapps/teams/api.py

```diff
@@ -264,16 +265,34 @@ def get_team_count_query_set(topic_id_set, course_id, organization_protection_st
         filter_query.update(
             {'organization_protected': organization_protection_status == OrganizationProtectionStatus.protected}
         )
-    return CourseTeam.objects.filter(**filter_query)
+    return filter_query
 
 
-def add_team_count(topics, course_id, organization_protection_status):
+def get_teams_with_visibility(user, topic_id_set, course_id, organization_protection_status):
```

Comment:
nit: what about calling it: 

> get_teams_based_on_organization_protection

 or something along those lines to be more explicit ? While I understand that organization protection status drives visibility, devs aren't familiar with the domain could get confused.

Reply:
It goes a bit further than that since it also does public/private team analysis, instead of organization protection which is just the FERPA access.

What about `get_teams_accessible_by_user` or `get_teams_with_user_access`?

Reply:
get_teams_accessible_by_user sounds good!

Reply:
Or `get_visible_teams_for_user`?

Reply:
Updated, thanks @Andytr1 👍